forc/src/ops/forc_edit.rs

```diff
@@ -0,0 +1,67 @@
+use std::path::PathBuf;
+
+use crate::cli::{AddCommand, RemoveCommand};
+use anyhow::{bail, Result};
+use forc_pkg::manifest::{Dependency, DependencyDetails, ManifestFile};
+
+pub fn add(
+    /*the command from the user that is typed into the terminal*/ command: AddCommand,
+) -> Result<()> {
+    // 1. How will forc know where to add a dependency to?
+
+    // type of the variable
+    let AddCommand {
+        dependency,    // whatever library the user is trying to add to the toml
+        manifest_path, // the path to forc.toml
+    } = command; // variable that we got from the add function
+
+    // matches the path to the forc.toml
+    let dir = match manifest_path {
+        Some(ref path) => PathBuf::from(path), // if there is _some_ manifest_path, then store the manifest_path
+        None => std::env::current_dir()?,      // if not, then use the current directory
+    };
+    let manifest = ManifestFile::from_dir(&dir)?; //stores the forc.toml in a varible, and checks if it is a package or a workspace
+    let pkg_manifest = if let ManifestFile::Package(pkg_manifest) = &manifest {
+        // if the manifest is a package...
+        pkg_manifest // store this package
+    } else {
+        bail!("forc-edit does not support workspaces.") // otherwise bail
+    };
+
+    let dependency_path = PathBuf::from(&dependency); // create a path buffer from the dependency string
+    let dep_pkg = if let Ok(dep_path) = ManifestFile::from_dir(&dependency_path) {
+        // if the dependency path exists...
+        if let ManifestFile::Package(dep_pkg) = dep_path {
+            // ...store this variable
+            dep_pkg
+        } else {
+            bail!("forc-edit does not support workspaces.") // otherwise bail
+        }
+    } else {
+        bail!("dependency path does not contain a forc.toml") // otherwise bail
+    };
+    let key = dep_pkg.project.name.clone(); // the key is the name of the project
+    let value = Dependency::Detailed(DependencyDetails {
+        // the value is the path to the dependency
+        version: None,
+        path: Some(dependency),
+        git: None,
+        branch: None,
+        tag: None,
+        package: None,
+        rev: None,
+    });
+
+    // 2. How will we find the dependency block in the forc.toml?
+    //
+    // Todo: can I insert a new key value pair directly into this BTreeMap?
+    if let Some(mut deps) = pkg_manifest.dependencies.clone() {
+        // 3. Write new dependency to that block (name and path)
+        deps.insert(key, value);
+    }
```

Comment:
This currently clones the package manifest's dependencies set and inserts the dependency to the cloned dependencies set, but does nothing with the cloned dependencies. There are a few issues with this:

1. This doesn't update the original `pkg_manifest` instance. We can fix this with something like:
   ```rust
       if let Some(deps) = pkg_manifest.dependencies.as_mut() {
           deps.insert(key, value);
       }
   ```
2. We should also check whether or not the dependency already exists first. E.g. if the user already has the dependency entry under their dependencies, we probably don't want to implicitly override an existing entry that might point to an alternative source.
3. We don't write the result to the manifest file. Even if we do make the change mentioned in `1.`, we still need to write the updated manifest to the user's `Forc.toml` file. We should do so in a way that preserves the original formatting and layout of the manifest too, so we might need to consider using a crate like `toml_edit` for this (as mentioned [here](https://github.com/FuelLabs/sway/issues/2369#issuecomment-1193609370)).